lib/create-options.js

```diff
@@ -9,7 +9,16 @@ function createOptions(cwd, context) {
   }
   options.env = Object.create(process.env);
   options.env['npm_loglevel'] = 'error';
-  options.env['npm_config_tmp'] = context.npmConfigTmp;
+  // Overwrite the npm temp directory if not already set via the env variable
+  if (!options.env['npm_config_tmp']) {
```

Comment:
all the logic for the should be in `temp-directory.js` there should be no need for any checks or alternative code paths at this point